maas/client/viscera/machines.py

```diff
@@ -389,6 +390,10 @@ class Machine(Object, metaclass=MachineType):
                 raise RescueModeFailure(msg, machine)
             return machine
 
+    async def get_details(cls):
+        data = await cls._handler.details(system_id=cls.system_id)
+        return bson.loads(data)
+
```

Comment:
This should be an instance method not a cls method. You need to rename `cls` to `self`, because in this case it really is `self`.